### ISSUE TITLE
AI Dungeon-like interface mode

### ISSUE THREAD

**Issue by arch-official** [oobabooga/text-generation-webui#649]
**Proposal**

Since AI Dungeon can be very limiting and not very coherent, I think a lot of people would like to have a similar interface to AI Dungeon but with the ability to run their own AI models. Pairing this interface with KoboldAI Erebys or Nerybus Mix would give a similar feel to running AI Dungeon, except locally and not on someone else's server.

Also using a similar function to the chat interface modes using the log file to keep a history of the story would be the best way to go as the feature is already built in.

**Additional Information**

Specifically, the ability to use the "Story", "Do", "Say" and "Edit" buttons shown here.

![image](https://user-images.githubusercontent.com/116774094/228660266-1495f7a7-55e4-4bca-b461-3bfb81eb7d21.png)
![image](https://user-images.githubusercontent.com/116774094/228660371-0ddef66a-97ea-45e6-98e2-f74e8474a776.png)
![image](https://user-images.githubusercontent.com/116774094/228661133-852e40cd-4fa0-4bfb-b630-e06d6ce4cb7f.png)

This would undoubtedly rival and dominate AI Dungeon as we have the ability to use our own models instead of using their now outdated and limited models.

P.S.
Absolutely loving this text gen software, keep up the great work!

**Comment by Titaniumtown** [comment]
Yes that would be very fun to have. I have been creating my own prompt for something like this with LLaMA and it hasn't worked so well. Having this all integrated would be very useful.

**Comment by eiery** [comment]
In the background what exactly do the "Story", "Do", and "Say" buttons add to the prompt? I imagine they prefix something to your message.

**Comment by arch-official** [comment]
"Story" allows you to write in anything, adding to the story without explicitly being a "Do" or "Say" command, while being able to add key elements to your story that you wish to add.

Input:
![image](https://user-images.githubusercontent.com/116774094/229118287-d18d6f28-d5ba-48b8-99a4-544168cd460b.png)
Output:
![image](https://user-images.githubusercontent.com/116774094/229118584-9741dd95-3a7a-492a-8381-6f9edfe89580.png)

"Do" is a command to have specifically the players character do something, giving direct instruction for only the player's character.

Input:
![image](https://user-images.githubusercontent.com/116774094/229118968-e531c8e0-d5b8-4fdc-9eed-05a7241c57fe.png)
Output:
![image](https://user-images.githubusercontent.com/116774094/229119122-ccd60930-6d69-492e-8bec-6e83174f2443.png)

"Say" makes the player's character say something that is written in the input field. (This is also AI Dungeon's weakest point for coherency)

Input:
![image](https://user-images.githubusercontent.com/116774094/229123530-c93bd27b-e6c9-41e7-b9e9-37dda5ce6e61.png)
Output:
![image](https://user-images.githubusercontent.com/116774094/229123682-d80dba30-ad49-4d22-a59e-4a1aa76da0ca.png)

This would be absolutely amazing if implemented in text-gen

**Comment by eiery** [comment]
I tried this prompt and syntax out in the UI in Notebook mode with the 2.8B Pythia model and the Spinx Moth preset. It worked okay and it looks like the addition of "You..." for Do and "You say" for Say make the model able to understand my intent.

The AI rambles a bit at times so I just hit stop and reformat the output as I would rather it say too much than too little. Bold text was provided as input and everything else was generated.

> **You are Arch, a noble living in the kingdom of Larion. You have a pouch of gold and a small dagger. You are awakened by one of your servants who tells you that your keep is under attack. You look out the window and see a group of knights, small in number but still a few, on their way to the keep. You also hear the sounds of men yelling and the sound of swords clashing. You grab your gold, run out of your chamber and out the back door of the keep. You hear the loud crunch of armor and weapons hitting stone as your enemies make their way inside. Your servant yells at you from behind, but you ignore him as you grab an arrow and nook it to the bow. Slowly, carefully, you walk out into the courtyard just as your enemies storm in. One of them raises his shield, halting your arrow in the string.**
> **You release your arrow at the invaders, piercing the shield with ease and dropping one of them dead in their tracks.**
> "Now what?" asks the other knight. "I'm sure there's more than enough gold here for us all."
> "No," says another knight. "We must kill this man before he kills us!"
> "But we're not going to let him live," says the first knight.
> The two knights draw their swords and charge towards you, swinging wildly at your body.
> **You say "Too bad."**
> and shoot both of them in the chest.
> They fall down lifelessly onto the ground.
> Your servant runs up to you and hands you some arrows.
> "Here they are," he says. "All yours."
> You grab the arrows and put them in your quiver.
> Then you turn around and head back into the keep.

I was unable to get this to work properly in Chat mode as it looks like the AI Dungeon style intends for the user and model to *collaboratively* write on the same page, rather then distinctly separating your messages and the AI's response. I would love to see an Adventure mode with the webui which can do all that.

**Comment by arch-official** [comment]
It seems like the Notebook mode is able to replicate the intended functions when prompted properly, which is very promising, so maybe the Adventure mode could be a slightly altered Notebook mode that automatically saves the progress of the story (Similar to Logs in chat mode) and also gives you a more immersive reading area like full page with scrollable text, similar to AI Dungeon.
It would be so amazing to be able to run something more powerful and immersive than AI Dungeon on my own computer, hopefully the devs agree!

**Comment by arch-official** [comment]
I would still like to see this feature added, does anyone know how to make new interfaces that could recommend how I can do this myself and contribute it to the main repo?
